tools/override-generator/main_test.go

```diff
@@ -0,0 +1,202 @@
+			)`,
+			want: `go_deps.gazelle_override(
+				build_extra_args = ["-go_naming_convention_external=go_default_library"],
+				build_file_generation = "on",
+				directives = ["gazelle:proto disable"],
+				path = "github.com/apache/thrift",
+			)`,
+		},
+        {
+            name: "module override and gazelle",
+            give:  `load("@bazel_gazelle//:deps.bzl", "go_repository")
+
+            go_repository(
+                name = "com_github_bazelbuild_bazel_watcher",
+                build_extra_args = ["-go_naming_convention_external=go_default_library"],
+                build_file_generation = "off",  # keep
+                build_file_proto_mode = "disable",
+                importpath = "github.com/bazelbuild/bazel-watcher",
+                patch_args = ["-p1"],
+                patches = [
+                    # Remove it after they release this PR https://github.com/bazelbuild/bazel-watcher/pull/627
+                    "//patches:com_github_bazelbuild_bazel_watcher-go-embed.patch",
+                ],
+                sum = "h1:EfJzkMxJuNBGMVdEvkhiW7pAMwhaegbmAMaFCjLjyTw=",
+                version = "v0.23.7",
+            )`,
+            want: `go_deps.gazelle_override(
+                build_extra_args = ["-go_naming_convention_external=go_default_library"],
+                build_file_generation = "off",
+                directives = ["gazelle:proto disable"],
+                path = "github.com/bazelbuild/bazel-watcher",
+            )
+
+            go_deps.module_override(
+                patch_strip = 1,
+                patches = [
+                    #RemoveitaftertheyreleasethisPRhttps://github.com/bazelbuild/bazel-watcher/pull/627
+                    "//patches:com_github_bazelbuild_bazel_watcher-go-embed.patch",
+                ],
+                path = "github.com/bazelbuild/bazel-watcher",
+            )`,
+        },
+        {
+            name: "directives and proto args",
+            give: `go_repository(
+                name = "com_github_clickhouse_clickhouse_go_v2",
+                build_directives = [
+                    "gazelle:resolve go github.com/ClickHouse/clickhouse-go/v2/external @com_github_clickhouse_clickhouse_go_v2//external",
+                ],
+                build_extra_args = ["-go_naming_convention_external=go_default_library"],
+                build_file_generation = "on",
+                build_file_proto_mode = "disable",
+                importpath = "github.com/ClickHouse/clickhouse-go/v2",
+                sum = "h1:Nbl/NZwoM6LGJm7smNBgvtdr/rxjlIssSW3eG/Nmb9E=",
+                version = "v2.0.12",
+            )`,
+            want: `go_deps.gazelle_override(
+                build_extra_args = ["-go_naming_convention_external=go_default_library"],
+                build_file_generation = "on",
+                directives = [
+                    "gazelle:resolvegogithub.com/ClickHouse/clickhouse-go/v2/external@com_github_clickhouse_clickhouse_go_v2//external",
```

Comment:
This is also missing spaces.

Reply:
This was an issue with my test helper to try to compare the rules without spaces. I guess spaces are important in some situations, so I changed the helper to only trim leading and trailing spaces on each line.